.github/workflows/ci.yml

```diff
@@ -79,6 +79,8 @@ jobs:
     steps:
       - name: Check out code
         uses: actions/checkout@v4
+      - name: Fetch tags
```

Comment:
Callout: @joe-elliott Since checkout GHA runs by default with with detached head and doesn't pull tags, I had to modify the CI for the test case I wrote to pass.
The directive: `with: fetch-tags: true` doesn't work because of an [issue](https://github.com/actions/checkout/issues/1471) on GHA, open from few years.

Reply:
It looks like [integration tests are failing](https://github.com/grafana/tempo/actions/runs/11192788791/job/31179779452?pr=4110#step:5:955) right now. Is that what you're referring to?

Why does this not work? I would expect GHA to use the changes in the branch when running CI.

Reply:
yup, the test case that failed is the one I added. The shell script's file permission change wasn't added so it failed. Added the change and tested out the GHA on my fork. The test should pass now.